src/abstractdataframe/prettytables.jl

```diff
@@ -0,0 +1,62 @@
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+#
+# Description
+# ==============================================================================
+#
+#     This file contains the definitions related to the interface with
+#     PrettyTables.jl
+#
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+# Default DataFrames highlighter for text backend.
+#
+# This highlighter changes the text color to gray in cells with `nothing`,
+# `missing`, `#undef`, and types related to DataFrames.jl.
+function _df_h_f(data,i,j)
+    try
+        return ismissing(data[i,j]) ||
+            data[i,j] == nothing ||
+            typeof(data[i,j]) <: Union{AbstractDataFrame, GroupedDataFrame,
+                                       DataFrameRow, DataFrameRows,
+                                       DataFrameColumns}
+    catch e
+        if isa(e, UndefRefError)
+            return true
+        else
+            rethrow(e)
+        end
+    end
+end
+
+const _DF_HIGHLIGHTER = Highlighter(_df_h_f, Crayon(foreground = :dark_gray))
+
+# Default DataFrames formatter for text backend.
+#
+# This formatter changes how the following types are presented when rendering
+# the data frame:
+#     - missing;
+#     - nothing;
+#     - Cells with types related to DataFrames.jl.
+
+function _df_formatter(v,i,j,truncstring = 32)
+    if typeof(v) <: Union{AbstractDataFrame, GroupedDataFrame, DataFrameRow,
+                          DataFrameRows, DataFrameColumns}
+
+        # Here, we must not use `print` or `show`. Otherwise, we can call
+        # `_pretty_table` to render the current table leading to a stack
+        # overflow.
+        str = sprint(ourshow, v, truncstring, context = :compact => true)
+        str = split(str, '\n')[1]
```

Comment:
why is this line needed?

Reply:
Does `ourshow` with `:compact => true` always returns a string without line breaks?

Reply:
I would say so. The corner case is:
```
julia> df = DataFrame("a\na" => 1, "b\nb"=>1)
1×2 DataFrame
│ Row │ a\na  │ b\nb  │
│     │ Int64 │ Int64 │
├─────┼───────┼───────┤
│ 1   │ 1     │ 1     │

julia> gdf = groupby(df, 1:2)
GroupedDataFrame with 1 group based on keys: a
a, b
b
First Group (1 row): a
a = 1, b
b = 1
│ Row │ a\na  │ b\nb  │
│     │ Int64 │ Int64 │
├─────┼───────┼───────┤
│ 1   │ 1     │ 1     │

julia> DataFrame(hard=[gdf, gdf])
2×1 DataFrame
│ Row │ hard                                                      │
│     │ GroupedDataFrame{DataFrame}                               │
├─────┼───────────────────────────────────────────────────────────┤
│ 1   │ GroupedDataFrame with 1 group based on keys: a\\na, b\\nb │
│ 2   │ GroupedDataFrame with 1 group based on keys: a\\na, b\\nb │
```
but even in this case currently we escaped this newline (can you please check this case?).

Otherwise we just use `summary` which should not produce newlines.

Reply:
Nice catch! Since `ourshow` escape `\n`, then `\\n` is passed to PrettyTables, that escape an escaped string leading to `\\\\n` :) I changed to `summary` to avoid this.

Reply:
Now, I get this:

```julia
julia> show(DataFrame(hard=[gdf, gdf]), truncate = 300)
2×1 DataFrame
 Row │ hard
     │ Grouped…
─────┼─────────────────────────────────────────────────────────
 1   │ GroupedDataFrame with 1 group based on keys: a\na, b\nb
 2   │ GroupedDataFrame with 1 group based on keys: a\na, b\nb
```

Is it right?

Reply:
This is perfect 👍